includes/libs/CSSMin.php

```diff
@@ -50,176 +54,474 @@ class CSSMin {
+			return $uri;
+		}
+
+		// A data URI couldn't be produced
+		return false;
+	}
+
+	/**
+	 * Serialize a string (escape and quote) for use as a CSS string value.
+	 * http://www.w3.org/TR/2013/WD-cssom-20131205/#serialize-a-string
+	 *
+	 * @param string $value
+	 * @return string
+	 * @throws Exception
+	 */
+	public static function serializeStringValue( $value ) {
+		if ( strstr( $value, "\0" ) ) {
+			throw new Exception( "Invalid character in CSS string" );
+		}
+		$value = strtr( $value, [ '\\' => '\\\\', '"' => '\\"' ] );
+		$value = preg_replace_callback( '/[\x01-\x1f\x7f-\x9f]/', function ( $match ) {
+			return '\\' . base_convert( ord( $match[0] ), 10, 16 ) . ' ';
+		}, $value );
+		return '"' . $value . '"';
+	}
+
+	/**
+	 * @param string $file
 	 * @return bool|string
 	 */
-	protected static function getMimeType( $file ) {
+	public static function getMimeType( $file ) {
+		// Infer the MIME-type from the file extension
+		$ext = strtolower( pathinfo( $file, PATHINFO_EXTENSION ) );
+		if ( isset( self::$mimeTypes[$ext] ) ) {
+			return self::$mimeTypes[$ext];
+		}
+
 		$realpath = realpath( $file );
-		// Try a couple of different ways to get the mime-type of a file, in order of
-		// preference
 		if (
 			$realpath
 			&& function_exists( 'finfo_file' )
 			&& function_exists( 'finfo_open' )
 			&& defined( 'FILEINFO_MIME_TYPE' )
 		) {
-			// As of PHP 5.3, this is how you get the mime-type of a file; it uses the Fileinfo
-			// PECL extension
 			return finfo_file( finfo_open( FILEINFO_MIME_TYPE ), $realpath );
-		} elseif ( function_exists( 'mime_content_type' ) ) {
-			// Before this was deprecated in PHP 5.3, this was how you got the mime-type of a file
-			return mime_content_type( $file );
-		} else {
-			// Worst-case scenario has happened, use the file extension to infer the mime-type
-			$ext = strtolower( pathinfo( $file, PATHINFO_EXTENSION ) );
-			if ( isset( self::$mimeTypes[$ext] ) ) {
-				return self::$mimeTypes[$ext];
-			}
 		}
+
 		return false;
 	}
 
 	/**
-	 * Remaps CSS URL paths and automatically embeds data URIs for URL rules
-	 * preceded by an /* @embed * / comment
+	 * Build a CSS 'url()' value for the given URL, quoting parentheses (and other funny characters)
+	 * and escaping quotes as necessary.
+	 *
+	 * See http://www.w3.org/TR/css-syntax-3/#consume-a-url-token
 	 *
-	 * @param $source string CSS data to remap
-	 * @param $local string File path where the source was read from
-	 * @param $remote string URL path to the file
-	 * @param $embedData bool If false, never do any data URI embedding, even if / * @embed * / is found
+	 * @param string $url URL to process
+	 * @return string 'url()' value, usually just `"url($url)"`, quoted/escaped if necessary
+	 */
+	public static function buildUrlValue( $url ) {
+		// The list below has been crafted to match URLs such as:
+		//   scheme://user@domain:port/~user/fi%20le.png?query=yes&really=y+s
+		//   data:image/png;base64,R0lGODlh/+==
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)